pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -256,6 +258,27 @@ type PagerDutyLinkConfig struct {
 	Text string `json:"alt,omitempty"`
 }
 
+// DiscordConfig configures notifications via Discord.
+// See https://prometheus.io/docs/alerting/latest/configuration/#discord_config
+type DiscordConfig struct {
+	// Whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The secret's key that contains the Discord webhook URL.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +optional
+	APIURL *v1.SecretKeySelector `json:"apiURL,omitempty"`
```

Comment:
this should be a required field? IIUC there's no global Discord URL parameter and Alertmanager will fail to load a discord config with an empty URL.

Reply:
Ref: https://prometheus.io/docs/alerting/latest/configuration/#configuration-file

Reply:
Strange, I can't seem to find discord configs in the docs, although I can see the support by reading Alertmanager's codebase.

Reply:
I guess the documentation on the website only shows the global config, and indeed Discord is not listed there:

https://github.com/prometheus/alertmanager/blob/a85979e19d24490322d5ce342301d17b0f13dcc5/config/config.go#L737-L767

https://github.com/prometheus/alertmanager/blob/a85979e19d24490322d5ce342301d17b0f13dcc5/config/config.go#L628-L644

Reply:
https://github.com/prometheus/alertmanager/issues/3360

It seems like it was supposed to be released but hasn't been yet.